peer-finder/peer-finder.go

```diff
@@ -62,11 +63,22 @@ func lookup(svcName string) (sets.String, error) {
 func shellOut(sendStdin, script string) {
 	log.Printf("execing: %v with stdin: %v", script, sendStdin)
 	// TODO: Switch to sending stdin from go
-	out, err := exec.Command("bash", "-c", fmt.Sprintf("echo -e '%v' | %v", sendStdin, script)).CombinedOutput()
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("echo -e '%v' | %v", sendStdin, script))
+
+	stdout, err := cmd.StdoutPipe()
```

Comment:
would `cmd.Output()` do this a lot more clearerly?

Reply:
No. Afaik, `cmd.Output()` will pause until the program terminates and then return it's output. This approach is streaming and will write the data to stdout as it happens (line-by-line). It allows to find problems with hanging on-start and on-change scripts that will never terminate (as it was the case for me).